pkg/util/names/names.go

```diff
@@ -45,17 +49,22 @@ func GetClusterName(executionSpaceName string) (string, error) {
 	return strings.TrimPrefix(executionSpaceName, executionSpacePrefix), nil
 }
 
-// GenerateBindingName will generate binding name by namespace, kind and name
+// GenerateBindingName will generate binding name by kind and name
 func GenerateBindingName(kind, name string) string {
 	return strings.ToLower(name + "-" + kind)
 }
 
-// GenerateWorkName will generate work name by namespace, kind and name
+// GenerateWorkName will generate work name by hash of its namespace, kind and name.
 func GenerateWorkName(kind, name, namespace string) string {
+	var workName string
 	if len(namespace) == 0 {
-		return strings.ToLower(name + "-" + kind)
+		workName = strings.ToLower(name + "-" + kind)
+	} else {
+		workName = strings.ToLower(namespace + "-" + name + "-" + kind)
 	}
-	return strings.ToLower(namespace + "-" + name + "-" + kind)
+	hash := fnv.New32a()
+	hashutil.DeepHashObject(hash, workName)
+	return rand.SafeEncodeString(fmt.Sprint(hash.Sum32()))
```

Comment:
What's the `work` name look like with this patch? Can you give an example?

Reply:
For example, the work name `default-b0d72d2a-modelpool`'s generated name is `774fd95bcf`. It is a `uint32` hash code and safely encoded. The origin uint32 hash code is `3309851679`.